wgengine/magicsock/magicsock_test.go

```diff
@@ -542,6 +543,140 @@ func TestDeviceStartStop(t *testing.T) {
 	dev.Close()
 }
 
+type testConnClosingContext struct {
+	parent context.Context
+	mu     *sync.Mutex
+}
+
+func (c *testConnClosingContext) Deadline() (deadline time.Time, ok bool) {
+	d, o := c.parent.Deadline()
+	return d, o
+}
+func (c *testConnClosingContext) Done() <-chan struct{} {
+	return c.parent.Done()
+}
+func (c *testConnClosingContext) Err() error {
+	// Deliberately deadlock if anything grabs the lock after checking Err()
```

Comment:
maybe panic instead of the second call? would be more explicit to debug later, as it's totally reasonable to expect future refactoring of magicsock to think it's able to call Err twice, per the normal context contract.

Reply:
I don't understand quite what this means; do you propose to add an integer counter to the testConnClosingContext and have this be:

```
func (c *testConnClosingContext) Err() error {
        if (c.count > 0) {
                panic('...')
        }
        return errors.New("testConnClosingContext error")
}
```